util/flask_util.py

```diff
@@ -23,15 +26,17 @@ def problem(type, status, title, detail=None, instance=None, headers={}):
 def languages_for_request():
     return languages_from_accept(flask.request.accept_languages)
 
-def originating_ip():
-    """If there is an X-Forwarded-For header, use its value as the
-    originating IP address. Otherwise, use the address that originated
-    this request.
+def originating_ip() -> str:
+    """Determine the client's IP address.
+
+    If there is an X-Forwarded-For header and it has a non-empty value,
+    use the client value as the originating IP address. Otherwise, use
+    the address that originated this request.
+
+    NB: The format of an X-Forwarded-For header value is: "<client-ip>, [proxy1-ip, [proxy2ip ...]]
+
+    :return: IP address of request originator
+    :rtype: str
     """
-    address = None
-    header = 'X-Forwarded-For'
-    if header in flask.request.headers:
-        address = flask.request.headers[header]
-    if not address:
-        address = flask.request.remote_addr
-    return address
+    addresses = re.split(_COMMA_SPACE_SEPARATOR, flask.request.headers.get('X-Forwarded-For', '').strip())
```

Comment:
This is just a minor nit, nothing worth holding this PR up, but instead of using a regular expression to do the splitting, wouldn't is be easier to use the string `split` function? 

Something like:
```python
addresses = flask.request.headers.get('X-Forwarded-For', '').split(',')
```

Reply:
I see why you are using a regex now. Using a string split doesn't remove the potential white space surrounding the IP. 

We could use use a `map` with `str.strip`. Something like:
```python
addresses = map(str.strip, flask.request.headers.get('X-Forwarded-For', '').split(','))
```

I'm not sure if that is more readable then just using a regex or not 🤷

Reply:
I considered this one, but went with the regex. I suppose this approach is a little cleaner. I could eliminate a module variable and an import.